LogKitten.java

```diff
@@ -152,34 +212,76 @@ private static void reportErrorToDriverStation(String details, String errorMessa
 	public static synchronized void logMessage(Object message, KittenLevel level, boolean override) {
 		message = message.toString(); // Not strictly needed, but good practice
 		if (LogKitten.logLevel.compareTo(level) >= 0) {
-			String content = LogKitten.timestamp() + " " + level.getName() + ": " + LogKitten.getLoggerMethodCallerMethodName()
+			String content = getRobotMode() + " " + LogKitten.timestamp() + " " + level.getName() + ": " + LogKitten.getLoggerMethodCallerMethodName()
 				+ ": " + message + " \n";
 			try {
-				if (LogKitten.fileOutput != null) {
-					LogKitten.fileOutput.write(content.getBytes());
+				if (LogKitten.globalOutput != null) {
+					LogKitten.globalOutput.write(content.getBytes());
 				} else {
-					System.out.println("Error logging: logfile not open");
+					System.out.println("Error logging: global logfile not open");
 				}
 			}
 			catch (IOException ioe) {
 				System.out.println("Error logging " + level.getName() + " message");
 				ioe.printStackTrace();
 			}
 		}
+		for(Map.Entry<Kitten, String> i : kittenList.entrySet()) {
+			String content = getRobotMode() + " " + LogKitten.timestamp() + " " + level.getName() + ": " + LogKitten.getLoggerMethodCallerMethodName()
+			+ ": " + message + " \n";
+			switch(i.getValue()) {
+				case "RobotModeKitten":
+					if(getRobotMode() == ((RobotModeKitten)i.getKey()).getMode()) {
+						try {
```

Comment:
There's a lot of duplicate code between "RobotModeKitten" and "CANKitten" cases. The code:
```
	try {
 		if(i.getKey().output != null) {
 			i.getKey().output.write(content.getBytes());
 		} else {
 			System.out.println("Error logging: " + i.getKey().category + " logfile not open");
		}
 	}
 	catch(IOException ioe) {
 		System.out.println("Error logging " + level.getName() + " message");
 		ioe.printStackTrace();
 	}
```

Could probably be reduced to a private method, or maybe a better flow could be worked out here.

Reply:
Actually, it might be better instead to move the `logMessage` method to the `Kitten` class itself, and have `LogKitten` extend `Kitten` or something of that sort. Looking at a screen is making me nauseous right now but I can take a closer look tomorrow hopefully.